### PR TITLE
docs: update gopls related documentation

### PR DESCRIPTION
**Pull request by hyangah**
Update microsoft/vscode-go#3148

### CODE REVIEW

File: docs/Go-modules-support-in-Visual-Studio-Code.md [hunk 1]
```diff
@@ -31,11 +27,12 @@ Yes, you can and this is the path forward for module support in VS Code. Please
 
 ### Why is code navigation and code completion slow when using Go modules?
 
-Code navigation and code completion definitely works better when using the language server from Google. So, please give that a try. 
+Code navigation and code completion definitely works better when using `gopls`. So, please give that a try.
 
-If you are not using the language server, then this is mostly due to the limitation of the tools that power these features which are `godef` and `gocode` respectively. The Go tools team at Google are working on a [language server](https://godoc.org/golang.org/x/tools/cmd/gopls) which will be the long term solution for all language features.
+If you are not using the language server, then this is mostly due to the limitation of the tools that power these features which are `godef` and `gocode` respectively. The Go tools team at Google are working on a [language server](https://github.com/golang/tools/blob/master/gopls) which will be the long term solution for all language features.
```
Comment (stamblerre) [inline docs/Go-modules-support-in-Visual-Studio-Code.md:32]: maybe https://golang.org/s/gopls?

File: docs/Go-with-VS-Code-FAQ-and-Troubleshooting.md [hunk 2]
```diff
@@ -85,9 +85,29 @@ _Tip_: If you are looking for logs after doing a particular operation, first cle
...
+**A:** You may noticed this problem because of the window popup with a
+message such as "Running Save Participants ...". This indicates the
+underlying tools used for file formatting or other code action such as
+package imports take longer than desired. Before 1.42, VS Code aborted
+the code action and skipped formatting, which resulted in file saving
+without expected formatting or import. After 1.42, VS Code changed to
+surface the issue and let users either wait or cancel the entire file
+save operation.
+
+A workaround is either 1) disable "On Save" features, or 2) cancel the
```
Comment (stamblerre) [inline docs/Go-with-VS-Code-FAQ-and-Troubleshooting.md:99]: I'd rather not advertise disabling on save stuff? Maybe first we encourage people to file an issue and as a last resort they can disable.

File: docs/Features-for-Go-in-VS-Code.md [hunk 3]
```diff
@@ -4,20 +4,23 @@
 - Signature Help for functions as you type (using `gogetdoc` or `godef`+`godoc`)
 - Quick Info on the symbol as you hover over it (using `gogetdoc` or `godef`+`godoc`)
 
+If `go.useLanguageServer` is enabled, most of the above features will be
+offered through the Go language server `gopls`.
```
Comment (ramya-rao-a) [inline docs/Features-for-Go-in-VS-Code.md:8]: Why just above? Wouldnt some of the below features be offered via gopls as well?

File: docs/Features-for-Go-in-VS-Code.md [hunk 4]
```diff
@@ -4,20 +4,23 @@
...
+
 ### Code Navigation
 
-- Go to or Peek Definition of symbols (using `gogetdoc` or `godef`+`godoc`)
-- Find References of symbols and Implementations of interfaces (using `guru`)
-- Go to symbol in file or see the file outline (using `go-outline`)
-- Go to symbol in workspace (using `go-symbols`)
+- Go to or Peek Definition of symbols (using `gogetdoc` or `godef`+`godoc` or `gopls`)
```
Comment (ramya-rao-a) [inline docs/Features-for-Go-in-VS-Code.md:12]: How come such addition is not done in the `IntelliSense` section?

File: docs/Features-for-Go-in-VS-Code.md [hunk 5]
```diff
@@ -4,20 +4,23 @@
...
 - Toggle between a Go program and the corresponding test file.
 
 ### Code Editing
 
 - Code Snippets for quick coding
-- Format code on file save as well as format manually (using `goreturns` or `goimports` or `gofmt`)
-- Symbol Rename (using `gorename`. Note: For Undo after rename to work in Windows you need to have `diff` tool in your path)
-- Add Imports to current file (using `gopkgs`)
+- Format code on file save as well as format manually (using `goreturns` or `goimports` or `gofmt` or `gopls`)
+- Symbol Rename (using `gorename` or `gopls`. Note: For Undo after rename to work in Windows you need to have `diff` tool in your path)
```
Comment (ramya-rao-a) [inline docs/Features-for-Go-in-VS-Code.md:22]:
```suggestion
- Symbol Rename (using `gorename` or `gopls`. Note: If not using `gopls`, then for undo after rename to work in Windows you need to have `diff` tool in your path)
```

File: docs/Go-modules-support-in-Visual-Studio-Code.md [hunk 6]
```diff
@@ -1,24 +1,20 @@
...
+Add the below in your [settings](https://code.visualstudio.com/docs/getstarted/settings) to use Go modules.
 
 ```json
 "go.useLanguageServer": true
 ```
 
 > **Note**: You will be prompted to install the latest stable version of `gopls` as and when the Go tools team tag a new version as stable.
 
-Known issues in VS Code when using `gopls`:
+Extra settings to fine tune `gopls` are available. Please see the [documentation](https://github.com/golang/tools/blob/master/gopls/doc/vscode.md) for more information and recommended settings.
```
Comment (ramya-rao-a) [inline docs/Go-modules-support-in-Visual-Studio-Code.md:11]: Link text having more context always helps screenreaders

```suggestion
Extra settings to fine tune `gopls` are available. Please see the [gopls documentation](https://github.com/golang/tools/blob/master/gopls/doc/vscode.md) for more information and recommended settings.
```

File: docs/Go-with-VS-Code-FAQ-and-Troubleshooting.md (hunk 2 above)
Comment (hyangah) [reply docs/Go-with-VS-Code-FAQ-and-Troubleshooting.md]: PTAL.

File: docs/Features-for-Go-in-VS-Code.md (hunk 4 above)
Comment (hyangah) [reply docs/Features-for-Go-in-VS-Code.md:12]: I changed IntelliSense section to just explicitly mention `gopls` for each entry.

**Comment by ramya-rao-a** [comment]
@stamblerre Waiting for your approval before merging

**Pull request merged**